proxy/http/Http1ClientSession.cc

```diff
@@ -373,11 +367,7 @@ Http1ClientSession::state_keep_alive(int event, void *data)
     break;
 
   case VC_EVENT_EOS:
-    this->do_io_close();
-    if (_vc != nullptr) {
-      _vc->do_io_close();
-      _vc = nullptr;
-    }
+    this->do_io_close(EHTTP_ERROR);
```

Comment:
Shouldn't `EOS` not be an error?

Reply:
Explained in person - this is to force full close, never half close. Might be better to not pass an error and explicitly set `half_close = false;` but it's a minor point.